packages/block-editor/src/components/list-view/use-list-view-scroll-into-view.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+export default function useListViewScrollIntoView( {
+	firstSelectedBlockClientId,
+	numBlocksSelected,
+	selectedItemRef,
+} ) {
+	useEffect( () => {
+		if (
+			numBlocksSelected !== 1 ||
```

Comment:
What's the reason for the early return when there's a multiselection?

Reply:
This was mostly to try to simplify the approach — when a multiselection is applied, we can't know (in this implementation) whether or not it was done from the list view, in which case we wouldn't want the list view to be scrolled. For example, if a user selects a block, scrolls down the list view and then shift-selects a block further down, we don't want the list view to suddenly scroll back up to the initially selected block.

I imagined the main use case for scrolling the list view is changing single block selection, but happy to dig in further if you think we should handle multiselection from the outset.

Reply:
Ah, cool, that does sound like a good reason to start with single block selection. It'd be good to add a comment to that effect.